presto-main/src/main/java/com/facebook/presto/sql/analyzer/QueryExplainer.java

```diff
@@ -160,14 +160,14 @@ public String getJsonPlan(Session session, Statement statement, Type planType, L
 
         switch (planType) {
             case IO:
-                Plan plan = getLogicalPlan(session, statement, parameters, warningCollector);
+                Plan plan = getLogicalPlan(session, isRawQueryExplain, statement, parameters, warningCollector);
                 return textIOPlan(plan.getRoot(), metadata, session);
             default:
                 throw new PrestoException(NOT_SUPPORTED, format("Unsupported explain plan type %s for JSON format", planType));
         }
     }
 
-    public Plan getLogicalPlan(Session session, Statement statement, List<Expression> parameters, WarningCollector warningCollector)
+    public Plan getLogicalPlan(Session session, Boolean isRawQueryExplain, Statement statement, List<Expression> parameters, WarningCollector warningCollector)
```

Comment:
This doesn't have to be parameterized. Just create `LogicalPlanner` with `explain` always set to `true`.